python/kfserving/kfserving/api/kf_serving_client.py

```diff
@@ -167,6 +169,54 @@ def patch(self, name, kfservice, namespace=None, watch=False, timeout_seconds=60
         else:
             return outputs
 
+    def promote(self, name, namespace=None, watch=False, timeout_seconds=600): # pylint:disable=too-many-arguments,inconsistent-return-statements
```

Comment:
Promote as a concept seems a little strange to me. 

Does rollout make more sense?
rollout(0)
rollout(10)
rollout(50)
rollout() # same as 100. Promotes canary -> prod.

Reply:
@ellis-bigelow I think the `promote` is reasonable, the API is to move Canary Spec to Default Spec. For rollout, that's great requirement, we can add one more API calling `rollout` and let user input `canary_traffic_percent`.

Reply:
Seems reasonable to me, user might still want to keep default spec while 100 percent traffic is on canary in case they find a bug later on. So rollout 100 and promote can be separate steps.

Reply:
OK, will add a `rollout` api later, thanks.